### PR TITLE
Add name kwarg to all instruments

### LINKED ISSUE #658: Many instruments miss name argument
Many instruments miss a name argument and pass their name directly to the super constructor.

### PR DESCRIPTION
**Pull request by bmoneke**
All instruments have a default `name` kwarg. `test_all_instruments` is adjusted accordingly.

Closes #658

### CODE REVIEW

File: pymeasure/instruments/thorlabs/thorlabspm100usb.py [hunk 1]
```diff
@@ -34,11 +34,11 @@
...
     """Represents Thorlabs PM100USB powermeter."""
 
-    def __init__(self, adapter, **kwargs):
+    def __init__(self, adapter, name="ThorlabsPM100USB powermeter", **kwargs):
         super().__init__(
-            adapter, "ThorlabsPM100USB powermeter", **kwargs
+            adapter, name, **kwargs
         )
-        self.timout = 3000
+        self.timout = 3000  # TODO what is this for?
```
Comment (bilderbuchi) [inline pymeasure/instruments/thorlabs/thorlabspm100usb.py:41]: IDK, maybe @bleykauf remembers?

**Review by bilderbuchi — APPROVED** [review]
LGTM. Maybe we can resolve the TODO before merging.

File: pymeasure/instruments/thorlabs/thorlabspm100usb.py (hunk 1 above)
Comment (bmoneke) [reply pymeasure/instruments/thorlabs/thorlabspm100usb.py:41]: It was added by @dvspirito in the original addition of this instrument.

At that time, this (misspelled) property was not used either, so it was always an unused attribute.

I push a commit to remove this line (it is not used anyways).

Comment (bmoneke) [reply pymeasure/instruments/thorlabs/thorlabspm100usb.py:41]: @bilderbuchi feel free to commit my suggestion and merge.

Comment (bilderbuchi) [reply pymeasure/instruments/thorlabs/thorlabspm100usb.py]: Sounds good.

**Pull request merged**